src/category_theory/monad/equiv_mon.lean

```diff
@@ -0,0 +1,233 @@
+      intro X,
+      have := M.mul_one,
+      change (_ ◫ _) ≫ _ = _ at this,
+      rw [←nat_trans.id_hcomp_app, ←nat_trans.comp_app, this],
+      refl,
+    end,
+    right_unit' := begin
+      intro X,
+      have := M.one_mul,
+      change (_ ◫ _) ≫ _ = _ at this,
+      rw [←nat_trans.hcomp_id_app, ←nat_trans.comp_app, this],
+      refl,
+    end } }
+
+variable (C)
+/-- Passing from `Mon_ (C ⥤ C)` to `Monad C` is functorial. -/
+def Mon_to_Monad : Mon_ (C ⥤ C) ⥤ Monad C :=
+{ obj := of_Mon,
+  map := λ M N f,
+  { app_η' := begin
+      intro X,
+      simp only [auto_param_eq],
+      erw [←nat_trans.comp_app,f.one_hom],
+      refl,
+    end,
+    app_μ' := begin
+      intro X,
+      simp only [auto_param_eq],
+      erw [←nat_trans.comp_app, f.mul_hom],
+      simp only [nat_trans.naturality, assoc, nat_trans.comp_app],
+      erw [nat_trans.hcomp_app, assoc],
+      refl,
+    end,
+    ..f.hom } }
+variable {C}
+
+theorem of_to_mon_end_obj (M : Mon_ (C ⥤ C)) : (Mon_to_Monad C ⋙ Monad_to_Mon C).obj M = M :=
+  by {apply Mon_.hext, repeat {refl}}
+
+theorem to_of_mon_end_obj (M : Monad C) : (Monad_to_Mon C ⋙ Mon_to_Monad C).obj M = M :=
+  by {apply Monad.hext, repeat {refl}}
+
+theorem of_to_mon_end : Mon_to_Monad C ⋙ Monad_to_Mon C = 𝟭 _ :=
+begin
+  apply functor.ext,
+  { intros X Y f,
+    ext,
+    simp only [functor.id_map, functor.comp_map, Mon_.comp_hom', nat_trans.comp_app,
+      Mon_.hom_eq_to_hom, eq_to_hom_app, id_comp, eq_to_hom_refl, comp_id],
+    refl },
+  { intro X,
+    apply of_to_mon_end_obj },
+end
+
+theorem to_of_mon_end : Monad_to_Mon C ⋙ Mon_to_Monad _ = 𝟭 _ :=
+begin
+  apply functor.ext,
+  { intros X Y f,
+    ext,
+    simp only [Monad.comp_to_nat_trans, functor.id_map, functor.comp_map, nat_trans.vcomp_eq_comp,
+      nat_trans.comp_app, to_nat_trans_eq_to_hom, eq_to_hom_app, id_comp, eq_to_hom_refl, comp_id],
+    refl },
+  { intro X,
+    apply to_of_mon_end_obj }
+end
+
+/-- Isomorphism of functors used in `Monad_Mon_equiv` -/
+def of_to_mon_end_iso : Mon_to_Monad C ⋙ Monad_to_Mon C ≅ 𝟭 _ :=
+{ hom :=
+  { app := λ M,
+  { hom := 𝟙 _ } },
+  inv :=
+  { app := λ M,
+  { hom := 𝟙 _ } } }
+
+/-- Isomorphism of functors used in `Monad_Mon_equiv` -/
+def to_of_mon_end_iso : Monad_to_Mon C ⋙ Mon_to_Monad C ≅ 𝟭 _ :=
+{ hom :=
+  { app := λ M,
+  { app := λ X, 𝟙 _,
+    app_η' := by tidy,
+    app_μ' := by tidy } },
+  inv :=
+  { app := λ M,
+  { app := λ X, 𝟙 _,
+    app_η' := by tidy,
+    app_μ' := by tidy } } }
+
+variable (C)
+/-- Oh, monads are just monoids in the category of endofunctors (isomorphism of categories). -/
+def Monad_Mon_iso : (Cat.of $ Monad C) ≅ (Cat.of $ Mon_ (C ⥤ C)) :=
+{ hom := Monad_to_Mon C,
+  inv := Mon_to_Monad C,
+  hom_inv_id' := by apply to_of_mon_end,
+  inv_hom_id' := by apply of_to_mon_end }
+
+/-- Oh, monads are just monoids in the category of endofunctors (equivalence of categories). -/
+def Monad_Mon_equiv : (Monad C) ≌ (Mon_ (C ⥤ C)) :=
```

Comment:
```suggestion
@[simps]
def Monad_Mon_equiv : (Monad C) ≌ (Mon_ (C ⥤ C)) :=
```

Reply:
Done. I also added `@[simps]` to the other definitions in this file.